lib/puppet/type/mount.rb

```diff
@@ -293,5 +297,18 @@ def value(name)
       dependencies[0..-2]
     end
 
+    # Autorequire the mount point's file resource
+    autorequire(:file) { Pathname.new(self[:name]) }
+
+    # Autobefore the mount point's child file paths
+    autobefore(:file) do
+      dependencies = []
+      file_resources = catalog.resources.collect { |resource| resource.type == :file }
```

Comment:
You are absolutely correct. I tend to use `map` instead of `collect`, so I forgot what it does and blindly followed Josh's suggestion. I should have paid closer attention.